cpp/src/arrow/util/async_generator.h

```diff
@@ -65,6 +100,14 @@ Future<> VisitAsyncGenerator(AsyncGenerator<T> generator,
   return Loop(LoopBody{std::move(generator), std::move(visitor)});
 }
 
+/// \brief Waits for an async generator to complete, discarding results.
+template <typename T>
+Future<> AwaitAsyncGenerator(AsyncGenerator<T> generator) {
```

Comment:
I'm not the biggest fan of discard either as it implies to me the generator is doing nothing but I agree that "await" isn't very clear.  I've switched to discard.  This is used in the Scanner::ToTable method because the callbacks are side-effecting a table builder and after they've all run we can get a table out of the builder.